api/desecapi/models/records.py

```diff
@@ -56,24 +56,17 @@
 )
 
 
-def replace_ip_subnet(queryset, subnet):
+def replace_ip_subnet(records, subnet):
     """
-    Fetches A or AAAA record contents from an RRset queryset (depending on the subnet's
-    address family) and returns them with their subnet bits replaced accordingly.
+    Takes a list of A or AAAA records and returns them with their subnet bits replaced accordingly.
     """
-    subnet = ip_network(subnet, strict=False)
-    try:
-        records = queryset.get(
-            type={IPv4Network: "A", IPv6Network: "AAAA"}[type(subnet)]
-        ).records.all()
-    except ObjectDoesNotExist:
-        records = []
     return [
         str(
             ip_address(int(ip_address(record.content)) & int(subnet.hostmask))  # suffix
             + int(subnet.network_address)  # prefix
         )
         for record in records
+        if type(ip_address(record.content)) is type(subnet.network_address)
```

Comment:
Do not use `type` to compare types. Use `isinstance` instead. (https://docs.quantifiedcode.com/python-anti-patterns/readability/do_not_compare_types_use_isinstance.html)

`if isinstance(ip_address(record.content), type(subnet.network_address))`

Reply:
done